### PR TITLE
[SCB-2646] fix metaspace oom problem

### PR DESCRIPTION
**Pull request by yanghao605**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java [hunk 1]
```diff
@@ -205,7 +211,7 @@ private PriorityProperty<?> createIntProperty(Field field, String[] keys, Intege
 
   private String[] collectPropertyKeys(Field field, String prefix, Map<String, Object> parameters) {
     String propertyPrefix = prefix;
-    String[] keys = new String[] {field.getName()};
+    String[] keys = new String[]{field.getName()};
```
Comment (Shoothzj) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java:214]: checkstyle changes? Why not detected by checktsyle?

Comment (yanghao605) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java:214]: should I add a space?

Comment (liubao68) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java:214]: use formatter in etc foler and then format code

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java [hunk 2]
```diff
@@ -95,7 +100,8 @@ public List<ConfigObjectProperty> createProperties(Object instance, String prefi
         continue;
       }
 
-      Setter<Object, Object> setter = createObjectSetter(propertyDefinition);
+      Setter<Object, Object> setter = beanDescriptionCache.computeIfAbsent(propertyDefinition,
+              LambdaMetafactoryUtils::createObjectSetter);
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/ConfigObjectFactory.java:104]: import static org.apache.servicecomb.foundation.common.utils.LambdaMetafactoryUtils.createObjectSetter  can be removed

**Pull request merged**